### ISSUE TITLE
TabNavigator nested inside StackNavigator, does not render on moving quickly between screen and changing tab 

### ISSUE THREAD

**Issue by ravirajn22** [react-navigation/react-navigation#3693]
My App Stack is like,
```
const Root = StackNavigator({
  Tabs: TabNavigator({
    Tab1,
    Tab2
  }),
  NormalScreen1,
  NormalScreen2
});
```
Everything renders properly, but issue happens in the below scenario.

I have a button in Tab1, which navigates to NormalScreen1. If from NormalScreen1, I press the back button on the screen and go back to the TabNavigator i.e Tab1 and quickly navigate to Tab2. The Tab2 is not rendered, all I get is a plain white screen. Also when this happens the NormalScreen1 is mounted and unmounted unnecessary again, which could also be the reason for Tab2 not rendering. 

**NOTE:** This happens only on iOS Device (cant reproduce in the emulator)

Here is the Snack: https://snack.expo.io/H1U9cq0OM     

### Current Behavior

- Tab not rendering
-  Unnecessary screen mounting and unmounting

### Expected Behavior

- Tab should render
- No Unnecessary screen mounting and unmounting

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |v1.4.0
| react-native     |0.53.0
| node             |
| npm or yarn      |

Also, I have another issue with this version of react-navigation, where sometimes on switching tabs the whole screen is rendered from the bottom like a Modal instead of normal tab switching. But I could not reproduce this case exactly, so I am testing for the exact case. This also happens only on iOS.

**Comment by ravirajn22** [comment]
#3592

**Comment by matthamil** [comment]
Hi! Let us know when you can work up a reproducible example. This will help us to pinpoint the problem, otherwise there isn't much we can do.

**Comment by ravirajn22** [comment]
I have provided the snack you can use that to reproduce the first case.

**Comment by ericvicenti** [comment]
I've confirmed the bug on the snack you provided. Thanks for the report, will try to get this resolved soon

**Comment by darkredz** [comment]
same thing encountered, tabs not render on swiped but will render after a tab button click.

**Comment by lgraubner** [comment]
Having the same Problem.

-TabNavigator inside StackNavigator
- Navigating in the StackNavigator
- Navigating to a Tab
=> Tab unmounts

**Comment by AleksandrChernyavenko** [comment]
I found out temporary workaround. Need add next params to`TabNavigator` config fix this problem. 
```javascript
{
    initialRouteName: 'HomeTab',
    tabBarPosition: 'bottom',
    animationEnabled: true,       // add this to fix
    swipeEnabled: false,
    lazy: false,                  // add this to fix
    },
  }
``` 

( Looks like problem is in `ResourceSavingSceneView` -> wrong `visible` state ), first tab visit doesn't show content ( even `render` method of screen doesn't calls )
![GitHub Logo](https://lh5.googleusercontent.com/PSkzc_1DPjjsypmixuw1plwIPAdLglI1ZqY01uDWPsH6Gbt697eLFydYmBi1AYS4cF2-M91-7X2buLo8CkPS=w1413-h900)
!https://drive.google.com/file/d/1noU1gVKvBMm6oWvJaO0BjUtnfOW4QXoY/view?usp=sharing!

**Comment by lgraubner** [comment]
Thanks @AleksandrChernyavenko, the fix seems to work. Better than missing screens.

**Comment by lgraubner** [comment]
Just checked out v1.5.6 and this seems to fix the problem.

**Comment by ravirajn22** [comment]
Sadly I use `lazy: false` and have a lot of nested tabs. Making `lazy: true` will need a lot of work to fix, but let me try.

**Comment by AleksandrChernyavenko** [comment]
version 1.5.6 fixed problem with missing content in tab for me, but it has problem with tab position for iPhone 
![Tab-bar-iPhone](http://dl4.joxi.net/drive/2018/03/15/0024/2210/1624226/26/772c3f8e43.jpg)

**Comment by brentvatne** [comment]
update to 1.5.7 and that's resolved. there has been some thrashing on tab layout due to a regression in layout upstream. https://github.com/react-navigation/react-navigation/issues/3762

**Comment by malonguwa** [comment]
@brentvatne Thanks for the fix! does 1.5.7 has been released?

**Comment by brentvatne** [comment]
yes
